TextFormat.md

```diff
@@ -43,43 +38,377 @@ support more human-readable representations, but never at the cost of accurate r
 
 # Official Text Format
 
-WebAssembly currently doesn't have a final, official, text format. As detailed above the
-main purpose of the text format will be for human consumption, feedback from humans on
-readability will therefore factor into standardizing a text format.
-
-There are, however, prototype syntaxes which are used to bring up WebAssembly: it's easier
-to develop using a text format than it is with a binary format, even if the ultimate
-WebAssembly format will be binary. Most of these prototypes use [s-expressions][] because they
-can easily represent expression trees and [ASTs](AstSemantics.md) (as opposed to CFGs)
-and don't have much of a syntax to speak of (avoiding syntax bikeshed discussions).
-
-  [s-expressions]: https://en.wikipedia.org/wiki/S-expression
-
-Here are some of these prototypes. Keep in mind that these *aren't* official, and the final
-official format may look entirely different:
-
-* [Prototype specification][] consumes an s-expression syntax.
-* [WAVM backend][] consumes compatible s-expressions.
-* [sexpr-wasm prototype][] consumes compatible s-expressions, and works closely with the [V8 prototype][].
-* [LLVM backend][] (the `CHECK:` parts of these tests) emits compatible s-expressions.
-* [ilwasm][] emits compatible s-expressions.
-* [wassembler][] consumes a different syntax, and works closely with the [V8 prototype][].
-* [binaryen][] can consume compatible s-expressions.
-
-  [prototype specification]: https://github.com/WebAssembly/spec/tree/master/ml-proto/test
-  [LLVM backend]: https://github.com/llvm-mirror/llvm/tree/master/test/CodeGen/WebAssembly
-  [WAVM backend]: https://github.com/AndrewScheidecker/WAVM/tree/master/Test
-  [wassembler]: https://github.com/ncbray/wassembler/tree/master/demos
-  [V8 prototype]: https://github.com/WebAssembly/v8-native-prototype
-  [ilwasm]: https://github.com/WebAssembly/ilwasm
-  [sexpr-wasm prototype]: https://github.com/WebAssembly/sexpr-wasm-prototype
-  [binaryen]: https://github.com/WebAssembly/binaryen
+## Philosophy:
+
+ - Use JS-style sensibilities when there aren't reasons otherwise.
+ - It's a compiler target, not a programming language, but readability still counts.
+
+
+## High-level summary:
+
+ - Curly braces for function bodies, blocks, etc., `/* */`-style and `//`-style
+   comments, and whitespace is not significant.
+   Nested `/* */`-style comments follow JavaScript's rules.
+
+ - Semicolons separate expressions inside blocks, loops, and so on. The
+   last statement in the sequence may omit its trailing semicolon to indicate
+   that its result is being used as the result value of the parent construct.
+
+ - `get_local` looks like a simple reference; `set_local` looks like an
+   assignment. Constants use a simple literal syntax. This makes wasm's most
+   frequent opcodes very concise.
+
+ - Infix syntax for arithmetic, with simple overloading. Explicit grouping via
+   parentheses. Concise and familiar with JS and others.
+
```

Comment:
I think these are good points, however, if the code is isomorphic *except* for patterns like `unreachable + unreachable`, it might be better to lose information there, if that helps readability. I'm not saying that would be a pretty solution (e.g., "i32 if ambiguous"), but if it allows an overall better syntax, it might be worth it.

But on the other hand, I think `i32.(unreachable + unreachable)` is worth considering. Perhaps we could allow `i32.(unreachable + unreachable)` etc., and it would be necessary for ambiguous cases like this, but optional otherwise, so we can still write `$x + $y`. I could see the counterargument that the complexity is annoying though.

Side note, `i32.(operations)` might also be used to denote signed/unsigned, perhaps, `i32.(x / y)` vs `u32.(x / y)`?